### PR TITLE
DS-433 | @rebeccahongsf | Add error messages for paymentRefId use cases

### PR DESCRIPTION
**Pull request by rebeccahongsf**
# READY FOR REVIEW

# Summary
- Add error messages for paymentRefId use cases

# Review By (Date)
- preferably asap but by Dec 1 works as well.

# Review Tasks

## Setup tasks and/or behavior to test

1. Check out this branch on you local 
2. Enable the mockserver by adding `MEGAPROFILE_MOCK='true'` to your `.env`
3.  Run `npm run dev`
4. Navigate to `/membership/payment-history`
5. Open the `/src/utilities/mocks/memberships.js` and test the following use cases

### if user has payment ref id, the payment history form should display
1. Navigate to `/membership/payment-history` and confirm that the form displays.

### If the user has a membership but no associated payment ref id since they paid in full, the "Your membership plan has been paid in full" copy should appear
1. Within `memberships.js`, update the SAA membership `membershipGGPaymentReferenceID` to null (e.g. `membershipGGPaymentReferenceID: null`)
2. Navigate to `/membership/payment-history` and confirm that the correct copy displays.

### If user has no memberships, the "Interested in joining SAA?" copy should appear
1. Within `memberships.js`, comment out all memberships, leaving an empty array (e.g. `memberships: []`)
2. Navigate to `/membership/payment-history` and confirm that the correct copy displays.

### If user has a gsb membership but no saa membership, the "Interested in joining SAA?" copy should appear
1. Within `memberships.js`, comment back in the GSB membership obj, leaving only the SAA membership commented out.
2. Navigate to `/membership/payment-history` and confirm that the correct copy displays.

### Last but not least
1. Review code.

# Associated Issues and/or People
- DS-433

### CODE REVIEW

File: src/components/page-types/lightFormPage/lightFormPage.js [hunk 1]
```diff
@@ -83,8 +100,38 @@ const LightFormPage = (props) => {
...
     };
 
-    fetchData();
-  }, [orgId, dssId, paymentRefId]);
+    const findPaymentRefId = () => {
+      const saaMembership = memberships?.find(
+        (membership) => membership.membershipGroup === 'SAA'
+      );
+
+      console.log('SAA MEMBERSHIP: ', saaMembership);
```
Comment (sherakama) [inline src/components/page-types/lightFormPage/lightFormPage.js:108]:
```suggestion
```

File: src/components/page-types/lightFormPage/lightFormPage.js [hunk 2]
```diff
@@ -16,6 +16,39 @@ import { HeroImage } from '../../composite/HeroImage/HeroImage';
 import * as styles from './lightFormPage.styles';
 import { FlexBox } from '../../layout/FlexBox';
 
+const NoMembershipError = () => (
+  <div>
+    <Heading level={2} size="2" weight="semibold" className="su-rs-mb-3">
+      Interested in joining SAA?
+    </Heading>
+    <p className="su-card-paragraph">
+      <a href="/membership/join">Join now</a> or visit our{' '}
```
Comment (sherakama) [inline src/components/page-types/lightFormPage/lightFormPage.js:26]: We're preserving the UTMs in the SBLink utility component. Don't we need to do that here too?
https://github.com/SU-SWS/saa_alumni/blob/dev/src/utilities/sbLink.js#L128

**Comment by sherakama** [comment]
Screenshot 2023-11-29 at 10 25 09
Screenshot 2023-11-29 at 10 24 25
Screenshot 2023-11-29 at 10 24 20
Screenshot 2023-11-29 at 10 23 50
Screenshot 2023-11-29 at 10 23 14
Screenshot 2023-11-28 at 13 30 23

**Comment by sherakama** [comment]
Looks good to me. Can you look into the failed tests please?

**Comment by rebeccahongsf** [comment]
> Looks good to me. Can you look into the failed tests please?

@sherakama After some investigation, I realized that this PR sources data from the Storyblok copy I made of PROD (I wanted to have the latest and greatest for membership) so it doesn't include the same files as our Storyblok DEV environment. Because of the differences between PROD and DEV, not all tests pass with PROD data. 

Shall I create a ticket on syncing PROD and DEV along with updating the Cypress tests to our backlog?

**Comment by stfurkan** [comment]
> > Looks good to me. Can you look into the failed tests please?
> 
> @sherakama After some investigation, I realized that this PR sources data from the Storyblok copy I made of PROD (I wanted to have the latest and greatest for membership) so it doesn't include the same files as our Storyblok DEV environment. Because of the differences between PROD and DEV, not all tests pass with PROD data.
> 
> Shall I create a ticket on syncing PROD and DEV along with updating the Cypress tests to our backlog?

@rebeccahongsf I made some changes related to Cypress tests in https://github.com/SU-SWS/saa_alumni/pull/759 
If you merge `dev` branch to this PR, it might solve the test issue here :)

**Comment by rebeccahongsf** [comment]
> > > Looks good to me. Can you look into the failed tests please?
> > 
> > 
> > @sherakama After some investigation, I realized that this PR sources data from the Storyblok copy I made of PROD (I wanted to have the latest and greatest for membership) so it doesn't include the same files as our Storyblok DEV environment. Because of the differences between PROD and DEV, not all tests pass with PROD data.
> > Shall I create a ticket on syncing PROD and DEV along with updating the Cypress tests to our backlog?
> 
> @rebeccahongsf I made some changes related to Cypress tests in #759 If you merge `dev` branch to this PR, it might solve the test issue here :)

Ah, thanks for letting me know, @stfurkan ! 😃 I'll update my branches and try running the test again.

**Comment by stfurkan** [comment]
> > > > Looks good to me. Can you look into the failed tests please?
> > > 
> > > 
> > > @sherakama After some investigation, I realized that this PR sources data from the Storyblok copy I made of PROD (I wanted to have the latest and greatest for membership) so it doesn't include the same files as our Storyblok DEV environment. Because of the differences between PROD and DEV, not all tests pass with PROD data.
> > > Shall I create a ticket on syncing PROD and DEV along with updating the Cypress tests to our backlog?
> > 
> > 
> > @rebeccahongsf I made some changes related to Cypress tests in #759 If you merge `dev` branch to this PR, it might solve the test issue here :)
> 
> Ah, thanks for letting me know, @stfurkan ! 😃 I'll update my branches and try running the test again.

Seems like the changes in the other PR didn't fixed the test issues here :( Sorry @rebeccahongsf

**Comment by sherakama** [comment]
@rebeccahongsf, if there is anything you can do with the tests to guard against the content differences in prod/dev please do so. Otherwise, it has been a long time since we sync'd the content down from production. Please create a ticket for that in the backlog. We will need to make sure we don't upset the work our external developers are doing in that space prior to the sync.

**Comment by rebeccahongsf** [comment]
Sounds good @sherakama! I've created [DS-483](https://stanford.atlassian.net/browse/DS-483) for sync Storyblok Prod/Dev and ensure that the cypress tests still function. I'll look into whether we can adjust the tests to guard against the content differences today and follow up shortly.

**Comment by rebeccahongsf** [comment]
@sherakama Looks like we're able to configure [Cypress](https://docs.cypress.io/guides/references/configuration) to utilize specific [env variables](https://docs.cypress.io/guides/guides/environment-variables) for the tests. However, due to how we fetch environment variables from the vault, we'd need to specify the primary Storyblok Dev space (e.g., `CYPRESS_STORYBLOK_ACCESS_TOKEN`) to generate the correct test environment. But then this might potentially limit our ability to thoroughly test any additional functionalities introduced within the enhancement branch. 🤔 

For upcoming enhancements, we can establish a protocol where we consistently clone from StoryBlok Dev and ensure that Storyblok Dev remains synchronized with Production at regular intervals (e.g. every X amount of months). I'm pondering whether additional tests are necessary to safeguard against discrepancies between our production and development environments. If we maintain both spaces in sync as part of our standard practice, it might mitigate the need for specific tests to guard against inconsistencies. 

Do you have any further thoughts or suggestions on aspects I should explore before finalizing this PR? If anything, we can revisit this if the tests continue to fail for the membership merging into the dev PR. 😃

**Comment by sherakama** [comment]
Thanks @rebeccahongsf I'm going to merge this into membership and we can deal with the tests there.

**Pull request merged**